### PR TITLE
chore: add structured logs to location and fix permission handling (WPB-6358)

### PR DESCRIPTION
**Pull request by yamilmedina**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-6358" title="WPB-6358" target="_blank">TaskWPB-6358</a>  [Android] Location feature logs observability
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

- Add logs while getting location, so we can monitor for example; usages by gms, non-gms, location not enabled, etc.

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [3294](https://10.10.124.134/job/AR-build-pipeline/3294/) **failed**.

**Review by Garzas — APPROVED** [review]

File: app/src/main/kotlin/com/wire/android/AppLogger.kt [hunk 1]
```diff
@@ -19,16 +19,53 @@ package com.wire.android
...
+    ) = with(appLogger) {
+        val logJson = jsonStringKeyValues.toJsonElement()
+        val sanitizedLeadingMessage = if (leadingMessage.endsWith(":")) leadingMessage else "$leadingMessage:"
+        val logMessage = "$sanitizedLeadingMessage $logJson"
+        when (level) {
+            KaliumLogLevel.DEBUG -> d(logMessage)
+            KaliumLogLevel.INFO -> i(logMessage)
+            KaliumLogLevel.WARN -> w(logMessage)
+            KaliumLogLevel.ERROR -> e(logMessage, throwable = error)
+            KaliumLogLevel.DISABLED,
```
Comment (vitorhugods) [inline app/src/main/kotlin/com/wire/android/AppLogger.kt]: If it is disabled, we log anyway?

Comment (yamilmedina) [reply app/src/main/kotlin/com/wire/android/AppLogger.kt]: Good catch 👀

**Review by MohamadJaara — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [3303](https://10.10.124.134/job/AR-build-pipeline/3303/) **failed**.

**Pull request merged**